docs/architecture/ADR-003-borrow-assets.md

```diff
@@ -5,106 +5,114 @@
+One of the base functions of the Umee universal capital facility is to allow users to borrow allowed asset types, using their own uTokens (obtained normally, by depositing assets) as a collateral.
 
 ## Decision
 
 The Cosmos `x/bank` module and the existing `umee/x/leverage` deposit features are prerequisites for these new capabilities.
 
 The flow of events is as follows:
-- Borrower already has uTokens in their account
-- Borrower marks those uTokens as eligible for use as collateral, which stores them in the `x/leverage` module account
-- Borrower requests to borrow assets from `leverage` module - module checks request, disburses tokens if acceptable, and remembers borrow position
-- While the borrow position is open, transactions that would result in the ongoing account's borrow position being higher than its calculated borrow limit are prevented (i.e. borrowing too much, withdrawing too many uTokens that are being used as collateral, disabling essential collateral)
-- Eventually, the borrower pays the repayment amount (in full or in part) in the same asset denomination that was borrowed.
+
+- Borrower already has uTokens in their account.
+- Borrower marks his uTokens as eligible for collateral using the `x/leverage` module account.
+- Borrower requests to borrow assets from `x/leverage` module. The module validates the request, disburses tokens if acceptable, and stores borrow position.
+- While the borrow position is open, transactions that would result the borrow position being higher than its calculated borrow limit are prevented (i.e. borrowing too much, withdrawing too many uTokens that are being used as collateral, disabling collateral).
+- Eventually, the borrower repays the borrowed position (in full or in part).
 
 Additionally, the following events occur at `EndBlock`:
-- Repayment amounts for open borrow positions increases by a borrowed-token-specific interest rate
 
-The `umee/x/leverage` module will be responsible for remembering each open borrow position.
-If the same user account opens multiple borrow positions in the same token denomination, the second position simply increases the amount of the first.
+- Fess are added to the open borrow positions based on token-specific interest rate.
+
+The `umee/x/leverage` module stores each open borrow position.
+If the same user account opens multiple borrow positions in the same token, the second position simply increases the amount of the first.
 
-Additionally, rather than an account's specific uToken denoms being tied to specific borrow positions, the sum of all borrow positions and collateral uTokens related to an account is to calculate the account's borrow limit.
+Additionally, rather than segregating each borrow position with a specific collateral deposit (uToken coins) we aggregate them. The sum of all account's collateral uTokens related is used to calculate the account's borrow limit.
+We define a **borrow limit** invariant:
```

Comment:
```suggestion
We define a **borrow limit**:
```
Not truly an invariant, because interest and price movements can make it false.

Reply:
true, how about using `guard`?

Reply:
ok